src/main/java/seedu/address/model/module/ModuleLesson.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.model.module;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a lesson of a module.
+ * Guarantees: immutable; is valid as declared in {@link #isValidLesson(String)}
+ */
+public class ModuleLesson {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Lesson name should only contain alphanumeric characters and spaces, and it should not be blank.";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    private final String lessonName;
+
+    /**
+     * Creates and initialises a new {@code Lesson} object with a {@code lessonName}.
+     *
```

Comment:
I think `Lesson` should be `ModuleLesson`?

Reply:
Yes, it should be. Thanks for the note.